datastore/datastore.go

```diff
@@ -60,27 +102,47 @@ const (
 `
 )
 
+type hash struct {
+	Hash string `json:"hash"`
+}
+
+type downloadid struct {
+	Id        string `json:"id"`
+	Timestamp int64  `json:"timestamp"`
+	VFSPath   string `json:"vfs_path"`
+	DocType   string `json:"doc_type"`
+}
+
 func (self ElasticDatastore) GetSubject(
 	config_obj *config_proto.Config,
 	path api.DSPathSpec,
 	message proto.Message) error {
 
-	id := services.MakeId(path.AsClientPath())
-	data, err := services.GetElasticRecord(
-		self.ctx, config_obj.OrgId, "datastore", id)
-	if err != nil {
-		return err
+	//id := cvelo_services.MakeId(path.AsClientPath())
+	response, _, err := cvelo_services.QueryElasticRaw(self.ctx, config_obj.OrgId,
+		"transient", json.Format(get_download_id, path.AsClientPath()))
+	result := &downloadid{}
+	for _, hit := range response {
+		err = json.Unmarshal(hit, &result)
+		if err != nil {
+			continue
+		}
 	}
+	hits, _, err := cvelo_services.QueryElasticRaw(self.ctx, config_obj.OrgId,
+		"transient", json.Format(get_datastore_doc_query, result.Id))
 
 	record := &DatastoreRecord{
 		Timestamp: utils.GetTime().Now().UnixNano(),
 	}
-	err = json.Unmarshal(data, &record)
-	if err != nil {
-		return err
-	}
+	for _, hit := range hits {
+		err = json.Unmarshal(hit, &record)
+		if err != nil {
+			continue
+		}
 
-	return protojson.Unmarshal([]byte(record.JSONData), message)
+		return protojson.Unmarshal([]byte(record.JSONData), message)
+	}
+	return nil
 }
 
 func (self ElasticDatastore) SetSubject(
```

Comment:
You can not convert this sync function to async function without breaking a lot of assumptions.